### ISSUE TITLE
Werkzeug 0.11.6 breaks debug functionality on Mac OS X

### ISSUE THREAD

**Issue by cmeury** [pallets/werkzeug#895]
Running this code:

```python
import flask
app = flask.Flask(__name__)
app.debug = True
app.run()
````

with Flask 0.10.1 and Werkzeug 0.11.6 breaks on my system (see below). Downgrading to 0.11.5 remediates the issue.

````python
ced@cheliax ~/tmp/debug $ python3 run.py
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
 * Restarting with stat
 * Debugger is active!
Traceback (most recent call last):
  File "run.py", line 4, in <module>
    app.run()
  File "/usr/local/lib/python3.5/site-packages/flask/app.py", line 772, in run
    run_simple(host, port, self, **options)
  File "/usr/local/lib/python3.5/site-packages/werkzeug/serving.py", line 633, in run_simple
    application = DebuggedApplication(application, use_evalex)
  File "/usr/local/lib/python3.5/site-packages/werkzeug/debug/__init__.py", line 249, in __init__
    if self.pin is None:
  File "/usr/local/lib/python3.5/site-packages/werkzeug/debug/__init__.py", line 259, in _get_pin
    self._pin, self._pin_cookie = get_pin_and_cookie_name(self.app)
  File "/usr/local/lib/python3.5/site-packages/werkzeug/debug/__init__.py", line 159, in get_pin_and_cookie_name
    get_machine_id(),
  File "/usr/local/lib/python3.5/site-packages/werkzeug/debug/__init__.py", line 94, in get_machine_id
    _machine_id = rv = _generate()
  File "/usr/local/lib/python3.5/site-packages/werkzeug/debug/__init__.py", line 70, in _generate
    match = re.match(r'"serial-number" = <([^>]+)', dump)
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/re.py", line 163, in match
    return _compile(pattern, flags).match(string)
````

**Comment by thesquelched** [comment]
I'm also experiencing this same issue, python 3.5.1 on OS X.

**Comment by density** [comment]
I get a similar traceback and the last line is most informative:

```
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/re.py", line 163, in match
    return _compile(pattern, flags).match(string)
TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by untitaker** [comment]
Have you tried 0.11.7?

**Comment by cmeury** [comment]
Same thing happens with 0.11.7.

**Comment by taogeT** [comment]
@cmeury 

Could you please try to modify `werkzeug/debug/__init__.py` like this ?

From
`match = re.match(r'"serial-number" = <([^>]+)', dump)`

To
`match = re.match(r'"serial-number" = <([^>]+)', dump.decode())`

**Comment by cmeury** [comment]
The fix by @nickfrostatx works. Also, for completeness sake, the code change suggested by @taogeT solves the issue as well.
Thanks, everyone.

**Comment by untitaker** [comment]
Fixed in 0.11.8
